recipes/libdisasm/all/conanfile.py

```diff
@@ -0,0 +1,115 @@
+            self.build_requires("msys2/20190524")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = self.name + "-" + self.version
+        os.rename(extracted_folder, self._source_subfolder)
+
+    @contextmanager
+    def _build_context(self):
+        env = {}
+        if self.settings.compiler == "Visual Studio":
+            with tools.vcvars(self.settings):
+                env.update({
+                    "CC": "cl -nologo",
+                    "CXX": "cl -nologo",
+                    "CPP": "cl -E -nologo",
+                    "AR": "{} lib".format(self.deps_user_info["automake"].ar_lib.replace("\\", "/")),
+                    "LD": "link -nologo",
+                    "NM": "dumpbin -symbols",
+                    "STRIP": ":",
+                    "RANLIB": ":",
+                })
+                with tools.environment_append(env):
+                    yield
+        else:
+            yield
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        yes_no = lambda v: "yes" if v else "no"
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        self._autotools.libs = []
+
+        conf_args = [
+            "--enable-shared={}".format(yes_no(self.options.shared)),
+            "--enable-static={}".format(yes_no(not self.options.shared)),
+        ]
+        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        for patch in self.conan_data["patches"].get(self.version, []):
+            tools.patch(**patch)
+        with tools.chdir(self._source_subfolder):
+            self.run("autoreconf -fiv", run_environment=True, win_bash=tools.os_info.is_windows)
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.make()
+            if self.settings.os != "Windows":
+                autotools.make(args=["-C", "x86dis"])
+
+    def package(self):
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        with self._build_context():
+            autotools = self._configure_autotools()
+            autotools.install()
+            if self.settings.os != "Windows":
+                autotools.install(args=["-C", "x86dis"])
+
+        os.unlink(os.path.join(self.package_folder, "lib", "libdisasm.la"))
+
+    def package_info(self):
+        libname = "disasm"
+        if self.settings.os == "Windows" and self.options.shared:
+            libname += ".dll.lib" if self.settings.compiler == "Visual Studio" else ".dll.a"
```

Comment:
```suggestion
        if self.settings.compiler == "Visual Studio" and self.options.shared:
            libname += ".dll.lib"
```

Not sure, but adding `.dll.a` for MinGW should be useless.

Reply:
Shouldn't it at least add the `.dll`?

Reply:
There is nothig to disambiguate because static and shared are not packaged together.

Reply:
Thanks, I didn't know at the time that mingw does not require the `.a`.
But MSVC does. Doesn't it?

Assume a user has a dependency on a shared mingw libdisasm, and that the user has installed libdisasm.a in a public location (outside of conan). Won't it then use the wrong library?

Reply:
AFAIK, unless specificed, it links to shared lib.

Reply:
I would keep the `.dll` suffix.
Just to catch the error that `libdisasm.a` is available with a bad-behaving gcc/ld.

Reply:
If you keep .dll, it should be added in all recipes in CCI :s

Reply:
@SpaceIm 
Yes, it would be the safer approach.
It depends on what name the authors give the imp lib.

On the other hand, this will still error when the package builds a shared implib where a static library is wanted.

So it doesn't matter.

Maybe conan should automatically insert `-Bshared/-Bstatic`?

Reply:
might be fixed later if it breaks some build system (for example qt build breaks on windows if at least one lib in the dependency tree has a suffix with .dll.lib for mscv)